src/bodse/timetable/GTFS_download.py

```diff
@@ -0,0 +1,107 @@
+import requests
+from datetime import datetime, timedelta
+from pathlib import Path
+import zipfile
+import os
+from clint.textui import progress
+import sys
+
+
+def download_gtfs_feed(directory):
+    current_datetime = datetime.now()
+    current_date_time = current_datetime.strftime("%Y%m%d_%H-%M-%S")
+
+    try:
+        print("Downloading started")
+        url = "https://data.bus-data.dft.gov.uk/timetable/download/gtfs-file/all"
+
+        req = requests.get(url, stream=True)
+
+        filename = Path("GTFS_" + current_date_time + ".zip")
+        filepath = directory / filename
+
+        with open(filepath, "wb") as output_file:
+            total_length = int(req.headers.get("content-length"))
+            for chunk in progress.bar(
+                req.iter_content(chunk_size=1024),
+                expected_size=(total_length / 1024) + 1,
+            ):
+                if chunk:
+                    output_file.write(chunk)
+        print("Downloading Completed")
+
+        # Extract the downloaded zip file to a folder with the same name
+        extraction_folder = (
+            directory / filename.stem
+        )  # Using filename.stem to get the name without the extension
```

Comment:
Unnecessary comment explaining filename.stem, try to avoid comments which explain what a specific piece of code does and focus more on comments explaining why code does something or maybe more generic comments which summarise what a bunch of code is doing like the comment on line 33 ("Extract the...")